lib/core/asynctree.js

```diff
@@ -104,7 +104,9 @@ class AsyncTree extends EventEmitter{
   }
 
   shouldRejectNodePromise(err, node = this.currentNode) {
-    if ((err.isExpect || node.namespace === 'assert') && this.currentNode.isES6Async) {
+    const rejectNodeOnFailure = node.options?.rejectNodeOnFailure;
+    const abortOnFailure = err.abortOnFailure || Utils.isUndefined(err.abortOnFailure);
```

Comment:
We shouldn't define the logic for finding out `abortOnFailure` at two places, such things create a maintenance nightmare because if we are to update this logic in future, we would need to do it at two places, which is easy to miss. Also try to follow the DRY (Don't Repeat Yourself) principle.

Over here, we probably would need to get the `abortOnFailure` value as an argument to the function.

Reply:
Yeah makes sense. Okay